src/ensembl/src/content/app/browser/drawer/drawer-views/transcript-summary/TranscriptSummary.tsx

```diff
@@ -14,70 +14,320 @@
+      transcriptId: transcriptTrack?.stable_id,
+      genomeId: ensObjectGene.genome_id
+    },
+    skip: !transcriptTrack?.stable_id
+  });
 
-  if (!transcriptTrack) {
+  if (loading) {
     return null;
   }
 
+  if (!data?.gene || !data.transcript) {
+    return <div>No data available</div>;
+  }
+
+  const { gene, transcript } = data;
+  const defaultProductGeneratingContext = transcript.product_generating_contexts.find(
+    (entry) => entry.default
+  ) as ProductGeneratingContext;
+
+  const product = defaultProductGeneratingContext.product;
+  const stableId = getDisplayStableId(transcript);
+
+  const focusId = buildFocusIdForUrl({
+    type: 'gene',
+    objectId: gene.unversioned_stable_id as string
+  });
+
+  const entityViewerUrl = urlFor.entityViewer({
+    genomeId: ensObjectGene.genome_id,
+    entityId: focusId
+  });
+
+  const uniprotXref = product?.external_references.find(
+    (xref) => xref.source.id == 'Uniprot/SWISSPROT'
+  );
+
+  const ccdsXref = transcript.external_references.find(
+    (xref) => xref.source.id == 'CCDS'
+  );
+
+  const splicedRNALength = getCommaSeparatedNumber(
+    getSplicedRNALength(transcript)
+  );
+
+  const productAminoAcidLength = getCommaSeparatedNumber(
+    defaultProductGeneratingContext?.cds?.protein_length || 0
+  );
+
   return (
-    <div className={styles.drawerView}>
-      <div className={styles.container}>
+    <div className={styles.container}>
+      <div className={styles.standardLabelValue}>
         <div className={styles.label}>Transcript</div>
-        <div className={styles.details}>
-          <span className={styles.mainDetail}>{getTranscriptStableId()}</span>
-          <span className={styles.secondaryDetail}>
-            {transcriptTrack.additional_info}
+        <div className={styles.value}>
+          <div className={styles.featureDetails}>
+            <span className={styles.featureSymbol}>{stableId}</span>
+            {transcript.so_term && (
+              <span>{transcript.so_term.toLowerCase()}</span>
+            )}
+            {transcript.slice.strand.code && (
+              <span>{getStrandDisplayName(transcript.slice.strand.code)}</span>
+            )}
+            <span>{getFormattedLocation(ensObjectGene.location)}</span>
+          </div>
+        </div>
+      </div>
+
+      <div className={styles.standardLabelValue}>
+        <div className={styles.label}>Transcript name</div>
+        <div className={styles.value}>{transcript.symbol}</div>
+      </div>
+
+      <div className={styles.standardLabelValue}>
+        <div className={styles.label}>Transcript length</div>
+        <div className={styles.value}>{transcript.slice.location.length}</div>
+      </div>
+
+      <div className={styles.standardLabelValue}>
+        <div className={styles.value}>
+          {transcript.spliced_exons.length} exons
+        </div>
+      </div>
+
+      <div className={`${styles.standardLabelValue} ${styles.spaceAbove}`}>
```

Comment:
I wonder if this might not be a better substitution for the spaceAbove class:

```
<section>
  <div className="styles.row">
    <div className="styles.label">Blah</div>
    <div className="styles.value">Blah</div>
  </div>
</section>
```

the sections can then be spaced in the way we like

![image](https://user-images.githubusercontent.com/6834224/107394706-101f8e00-6af4-11eb-9146-ac6db01d3a87.png)

Reply:
I have renamed `standardLabelValue` to `row`. Do we need the enclosing `section` tags? We could just apply additional stylings to `row` to add the spaces.